Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -790,7 +790,7 @@ void HackStudioWidget::configure_project_tree_view()
 
     m_project_tree_view->on_selection_change = [this] {
         m_open_selected_action->set_enabled(!m_project_tree_view->selection().is_empty());
-        m_delete_action->set_enabled(!m_project_tree_view->selection().is_empty());
+        m_delete_action->set_enabled(!m_project_tree_view->selection().is_empty() && access(m_project->model().root_path().characters(), W_OK) == 0);
```

Comment:
What if the selected file is in a subdirectory for which the user does not have write permissions?

Reply:
Well, it depends on what your `root_path`(the project path) is.

_oh_.

Reply:
That's certainly better but doesn't quite deal with all possible cases. Here's another example:

1. Have a directory that user not user-writable and another one that _is_ user writable. Each contains a bunch of files.
2. Try to select two files, one from each of those directories.
3. Note how the `Delete` button is enabled but deletion fails:


https://user-images.githubusercontent.com/388571/125995375-383ea519-4d19-4b48-bc73-116a34b9117e.mp4